src/org/zaproxy/zap/extension/pscanrulesAlpha/XChromeLoggerDataInfoLeakScanner.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ * 
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ * 
+ * Copyright 2015 The ZAP Development Team
+ *  
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License. 
+ * You may obtain a copy of the License at 
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0 
+ *   
+ * Unless required by applicable law or agreed to in writing, software 
+ * distributed under the License is distributed on an "AS IS" BASIS, 
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+ * See the License for the specific language governing permissions and 
+ * limitations under the License. 
+ */
+
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import java.io.IOException;
+import java.util.Vector;
+
+import net.htmlparser.jericho.Source;
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+import sun.misc.BASE64Decoder;
+
+/**
+ * X-ChromeLogger-Data header information leak passive scan rule 
+ * @author kingthorin+owaspzap@gmail.com
+ */
+public class XChromeLoggerDataInfoLeakScanner extends PluginPassiveScanner{
+
+	private static final String MESSAGE_PREFIX = "pscanalpha.xchromeloggerdata.";
+	private static final int PLUGIN_ID = 10052;
+	
+	private PassiveScanThread parent = null;
+	private static final Logger logger = Logger.getLogger(XChromeLoggerDataInfoLeakScanner.class);
+	
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+	
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		long start = System.currentTimeMillis();
+	
+		Vector<String> xcldHeader = msg.getResponseHeader().getHeaders("X-ChromeLogger-Data"); //Get the header(s)
+		Vector<String> xcpdHeader = msg.getResponseHeader().getHeaders("X-ChromePhp-Data"); //Add any header(s) using the alternate name
+		Vector<String> allHeaders = new Vector<String>(2);
+		
+		try {
+			allHeaders.addAll(xcldHeader); //Load into a single vector
+		} catch (NullPointerException npe) {}
+		try {
+			allHeaders.addAll(xcpdHeader); //Load into a single vector
+		} catch (NullPointerException npe) {}
+		
+		if (allHeaders != null) { //Header Found

```

Comment:
When I caught the NPE I didn't leave (return), so if no relevant header was found I guess allHeaders could be null....new PR handles differently.